### ISSUE TITLE
IsEnabled doesn't update visual state after being updated with INotifyPropertyChanged when displaying shell alert on Windows 10

### ISSUE THREAD

**Issue by Laradius** [dotnet/maui#8877]
### Description

Visual state of button doesn't change after alert is displayed despite being enabled. It's visible as disable despite being clickable.

It works fine on Android.

![image](https://user-images.githubusercontent.com/31272990/180198867-9724cb39-807a-44b1-8089-bb28adcdeabc.png)

ObservableProperty and ICommand is part of Microsoft.Toolkit.Mvvm.ComponentModel

```cs
        [ObservableProperty]
        [AlsoNotifyChangeFor(nameof(IsNotBusy))]
        private bool isBusy;
        public bool IsNotBusy => !IsBusy;
```

```xml
<Button Text="DisplayCommand" Command="{Binding DisplayAlertCommand}" IsEnabled="{Binding IsNotBusy}">
```

```cs
[ICommand]
public async Task DisplayAlertAsync()
        {
            IsBusy = true;
            bool answer = await Shell.Current.DisplayAlert("Question?", "Would you like to play a game", "Yes", "No");
            IsBusy = false;
        }
```

### Steps to Reproduce

1. Run app on Windows 10
2. Click button
3. Visual state doesn't update after alert is displayed.

### Version with bug

6.0.419 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows 10

**Comment by Laradius** [comment]
https://github.com/Laradius/MauiErrorApp

**Comment by Laradius** [comment]
@Eilon @rmarinho Could I get informatiom regarding that issue?

**Comment by Eilon** [comment]
@Laradius we will check this out in an upcoming bug triage meeting. I'm wondering if the async behavior is somehow affecting this? I'm not sure how the MVVM Toolkit attributes work with this, either. What happens if you call `NotifyPropertyChanged(nameof(IsNotBusy))` after setting it to false?

**Comment by Laradius** [comment]
> @Laradius we will check this out in an upcoming bug triage meeting. I'm wondering if the async behavior is somehow affecting this? I'm not sure how the MVVM Toolkit attributes work with this, either. What happens if you call `NotifyPropertyChanged(nameof(IsNotBusy))` after setting it to false?

Is not busy property returns true and notifies all bindings about the change.

**Comment by Ahmadmansoor** [comment]
Hi
I have already posted this at another repo
https://github.com/CommunityToolkit/dotnet/issues/372
my problem is the button can't disable itself from the C# code.
and sometimes from the other buttons as well.
any guide pls, my whole work is stope at this stage.

**Comment by sude22** [comment]
Do anybody care about this issue?
Right now it is not possible to use Async Commands for buttons because the visual state of the button (from CanExecute)  may be not correct after showing a DisplayAlert. A showstopper in the moment for MAUI on Windows.....

**Comment by hartez** [comment]
Possibly related: https://github.com/dotnet/maui/issues/9753

**Comment by PureWeen** [comment]
Possibly related https://github.com/CommunityToolkit/dotnet/issues/316

**Comment by Sergio0694** [comment]
I've noticed the same in MAUI as well (it does indeed look related to #9753). This seems to be a MAUI specific issue and unrelated from the MVVM Toolkit per se (the same code works just fine on UWP and WinUI 3 too, for instance).

@Laradius unrelated, but you can simplify your code by removing those `IsBusy`/`IsNotBusy` properties entirely, and just bind to the `IsRunning` property exposed by the `IAsyncRelayCommand` property that the generator will create for you. For the inverted property (`IsNotBusy`), simply use a converter or `x:Bind` to function in XAML, you don't need an extra property 🙂

**Comment by hartez** [comment]
This was fixed by #11840.
